roles/docker/tasks/main.yml

```diff
@@ -22,4 +22,6 @@
 
 - name: restart docker
   shell: systemctl daemon-reload && systemctl stop docker && systemctl start docker-tcp.socket && systemctl start docker
-  when: docker_env_conf_result.changed or docker_tcp_conf_result.changed or docker_svc_conf_result.changed
+  when: docker_install_result.changed or swarm_install_result.changed or docker_env_conf_result.changed or
+        docker_tcp_conf_result.changed or docker_svc_conf_result.changed or force_restart
```

Comment:
ansible is a way to configure things and upgrade stuff just like any configuration management code. So, usually re-running same playbook should not have any side-effects if nothing changed. Adding convenience variable for a user overload this logic and make it complex to maintain.